### PR TITLE
many: remove the gadget yaml support from snappy

### PR DESCRIPTION
**Pull request by mvo5**
This branch removes all legacy gadget snap yaml handling.

### CODE REVIEW

File: snappy/install.go [hunk 1]
```diff
@@ -142,9 +142,8 @@ func NewConfiguredUbuntuStoreSnapRepository() *store.SnapUbuntuStoreRepository {
 	storeID := ""
 	if cand := os.Getenv("UBUNTU_STORE_ID"); cand != "" {
```
Comment (pedronis) [inline snappy/install.go]: can you add a // TODO set the store-id from the model information

**Comment by pedronis** [comment]
q: is allowGadget something that only u-d-f sets? if that's the case, can't we simplify the code in canInstall a bit more (with some TODO there if needed) and get rid of getGadget for a little while?

**Comment by pedronis** [comment]
no, actually see that is getGadget is used when !allowGadget, that is also afaict the last place it would be used, anyway I can implement it differently when I move canInstall

**Comment by pedronis** [comment]
LGTM, I suppose the real question is how images with this boot?

**Comment by mvo5** [comment]
@pedronis I just double checked, the image is still fine. The reasons this all works is because u-d-f contains its own yaml structs for the gadget snap. I build a new u-d-f against this branch, created a new test image (amd64) and ran the testsuite. It passes with no failures.

**Comment by mvo5** [comment]
Just to clarify, tested using a custom os snap that has  manually build/copied `snap` and `snapd` binaries.

**Comment by niemeyer** [comment]
Nice, thanks!

**Pull request merged**